### PR TITLE
Track2 complexgroup

### PR DESCRIPTION
**Pull request by catalinaperalta**
This PR adds code related to the reference samples for the following testserver endpoints:
basic_getValid
basic_putValid

Related to issues: #189 and #190

### CODE REVIEW

File: test/autorest/generated/complexgroup/internal/complexgroup/service.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+)
+
+// Service ..
+type Service struct{}
+
+// GetValidCreateRequest creates the GetValid request.
+func (Service) GetValidCreateRequest(u url.URL) (*azcore.Request, error) {
+	u.Path = path.Join(u.Path, "/complex/basic/valid")
```
Comment (jhendrixMSFT) [inline test/autorest/generated/complexgroup/internal/complexgroup/service.go]: The fact that this doesn't need an `api-version` query param is a bug in the [response file](https://github.com/Azure/autorest.testserver/blob/master/mappings/vanilla/complex_basic_valid-62466d5b-943d-47d3-a192-520d96d5d6c3.json#L5).  Can you please confirm?

File: test/autorest/generated/complexgroup/internal/complexgroup/models.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+func (e Error) Error() string {
+	return e.Message
+}
+
+// Basic ..
+type Basic struct {
+	ID    int    `json:"id,omitempty"`
+	Name  string `json:"name,omitempty"`
+	Color string `json:"color,omitempty"`
```
Comment (jhendrixMSFT) [inline test/autorest/generated/complexgroup/internal/complexgroup/models.go]: Color should be an enum.

File: test/autorest/generated/complexgroup/internal/complexgroup/models.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+// GetValidResponse ...
+type GetValidResponse struct {
+	StatusCode int
+	Value      Basic
```
Comment (jhendrixMSFT) [inline test/autorest/generated/complexgroup/internal/complexgroup/models.go]: The field name should be the same as the model name, so `Basic`.  We use `Value` only for operations that return scalar types.
Also, should be pointer-to-type.

File: test/autorest/generated/complexgroup/internal/complexgroup/service.go (hunk 1 above)
Comment (catalinaperalta) [reply test/autorest/generated/complexgroup/internal/complexgroup/service.go]: Yes, GetValid is currently able to respond without the api-version query parameter. PutValid on the other hand requires api-version in order to accept the request.

File: test/autorest/generated/complexgroup/internal/complexgroup/models.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+	return err
+}
+
+func (e Error) Error() string {
+	return e.Message
+}
+
+// Basic ..
+type Basic struct {
+	ID    int    `json:"id,omitempty"`
```
Comment (jhendrixMSFT) [inline test/autorest/generated/complexgroup/internal/complexgroup/models.go]: Fields need to be pointer-to-type, with the exception of maps and slices.  Same applies to `Error` as well (I missed this earlier).

File: test/autorest/generated/complexgroup/internal/complexgroup/models.go [hunk 5]
```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package complexgroup
+
+import "github.com/Azure/azure-sdk-for-go/sdk/azcore"
+
+// Error ...
+type Error struct {
```
Comment (jhendrixMSFT) [inline test/autorest/generated/complexgroup/internal/complexgroup/models.go]: We should add the correct JSON tags to the fields on this type (another thing I missed).

File: test/autorest/generated/complexgroup/internal/complexgroup/models.go (hunk 4 above)
Comment (catalinaperalta) [reply test/autorest/generated/complexgroup/internal/complexgroup/models.go]: You mean we need to have something like this?

// Basic ..
type Basic struct {
	ID    *int       `json:"id,omitempty"`
	Name  *string    `json:"name,omitempty"`
	Color *ColorType `json:"color,omitempty"`
}

File: test/autorest/generated/complexgroup/internal/complexgroup/models.go (hunk 2 above)
Comment (catalinaperalta) [reply test/autorest/generated/complexgroup/internal/complexgroup/models.go]: Done

File: test/autorest/generated/complexgroup/internal/complexgroup/models.go (hunk 3 above)
Comment (catalinaperalta) [reply test/autorest/generated/complexgroup/internal/complexgroup/models.go]: Good to know. I've updated it!

File: test/autorest/generated/complexgroup/internal/complexgroup/models.go (hunk 5 above)
Comment (catalinaperalta) [reply test/autorest/generated/complexgroup/internal/complexgroup/models.go]: Done. I added JSON tags to the error struct

File: test/autorest/generated/complexgroup/internal/complexgroup/models.go (hunk 4 above)
Comment (catalinaperalta) [reply test/autorest/generated/complexgroup/internal/complexgroup/models.go]: For now I'm assuming that you do mean what I put above and have pushed the changes. So also done?

File: test/autorest/generated/complexgroup/internal/complexgroup/models.go [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+	// ColorMagenta ...
+	ColorMagenta ColorType = "Magenta"
+	// ColorCyan ...
+	ColorCyan ColorType = "cyan"
+	// ColorBlack ...
+	ColorBlack ColorType = "blacK"
+)
+
+// ColorValues ...
+func ColorValues() []ColorType {
```
Comment (jhendrixMSFT) [inline test/autorest/generated/complexgroup/internal/complexgroup/models.go]: PossibleColorValues()

Comment (catalinaperalta) [reply test/autorest/generated/complexgroup/internal/complexgroup/models.go]: Done

**Pull request merged**